### PR TITLE
Added Modal Component

### PR DESCRIPTION
**Pull request by jyao244**
Finished Dialog and added a testing file for testing dialog.

### CODE REVIEW

**Comment by hajineats** [comment]
Could someone clarify if this is for the middle modal component here? 

Screen Shot 2022-03-11 at 1 57 13 AM

Would be nice if someone could clarify acceptance criteria for this (eg does this component need to contain logic for modal, or will there be a modal component?)

File: forgettable-frontend/src/components/Dialog/Dialog.js [hunk 1]
```diff
@@ -0,0 +1,16 @@
+import React from 'react';
+import classes from './Dialog.module.css';
+
+function Dialog(props) {
+  const {title, desc, cancalEvent, confirmEvent} = props;
```
Comment (hajineats) [inline forgettable-frontend/src/components/Dialog/Dialog.js:5]: Variable names should preferably be spelled in full as it helps other devs.

File: forgettable-frontend/src/components/Dialog/Dialog.js [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+  return <div className={classes.dialog}>
+    <div className={classes.title}>{title}</div>
+    <div className={classes.desc}>{desc}</div>
+    <div className={classes.toolbar}>
```
Comment (hajineats) [inline forgettable-frontend/src/components/Dialog/Dialog.js:9]: classes.toolbar is not in the Dialog.module.css file

File: forgettable-frontend/src/components/Dialog/Dialog.js [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+      <button className={classes.button1} onClick={cancalEvent}>Cancal</button>
+      <button className={classes.button2} onClick={confirmEvent}>Confirm</button>
```
Comment (hajineats) [inline forgettable-frontend/src/components/Dialog/Dialog.js:11]: These two buttons share the same style so reusing one of the css classes would be a better option here

File: forgettable-frontend/src/components/Dialog/Dialog.module.css [hunk 4]
```diff
@@ -0,0 +1,55 @@
+.dialog {
+  position: fixed;
+  background: white;
+  width: 525px;
+  height: 400px;
+  z-index: 2;
+  border-radius: 4px;
+  top: 50%;
+  left: 50%;
+  transform: translate(-50%, -50%);
```
Comment (hajineats) [inline forgettable-frontend/src/components/Dialog/Dialog.module.css:10]: There's a really useful mui (material ui) component called [Modal](https://mui.com/components/modal/#main-content). That will make css a lot more concise (no need to use fixed, top, left, translate, zindex as it automatically centers it for you!)

File: forgettable-frontend/src/components/Dialog/Dialog.module.css [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+  gap: 20px;
+  color: #000;
+}
+
+.title{
+  font-size: 30px;
+  color: brown;
+  margin-top: 50px;
+  margin-left: 123px;
+  margin-right: 123px;
```
Comment (hajineats) [inline forgettable-frontend/src/components/Dialog/Dialog.module.css:25]: `text-align: center` would be better here as we want to generally avoid hardcoding values

**Review by hajineats — CHANGES_REQUESTED** [review]
Hi Jiayou! Appreciate the time you took to create these changes. I've left comments here and there that might be helpful for you. Feel free to ping me if you need any help or clarification regarding these. Another side note, commented out codes should generally be avoided in production codes. Thanks!

**Comment by rainasong** [comment]
> Could someone clarify if this is for the modal component on the right?
> 
> Screen Shot 2022-03-11 at 1 57 13 AM
> 
> Would be nice if someone could clarify acceptance criteria for this (eg does this component need to contain logic for modal, or will there be a modal component?)
> 
> Also what should the title color/size be?

Thanks for bring this up and sorry if it's not clear in the issue.
The modal should cover all three cases below:
Screen Shot 2022-03-11 at 8 26 09 AM
- The title should be semibold, 30px (--text-xxlarge), *users should be able to specify the colour of the title. 
- The body of the modal should be configurable outside the modal component, as you can see we not only have cases where description is needed but also cases where we want buttons/input fields. This means users should be able to pass JSX/react component in as the children of the component from outside. (an example is as below)
Screen Shot 2022-03-11 at 8 41 31 AM
https://reactjs.org/docs/composition-vs-inheritance.html
- Users should be able to choose if they want 'Cancel' and 'Confirm' button or just 'Cancel' button.
- As described in this #64 issue here the modal should be able to close by clicking anywhere outside the modal area as well.
- I would recommend using the material UI modal component as the base and modify the styles as it will be easier and you can also add transitions and other effects later more easily. https://mui.com/components/modal/
(*User here means anyone who's using this component)

Please feel free to ask for further clarifications if you have any questions, and if anyone else has questions feel free to ask as well!

File: forgettable-frontend/src/components/Dialog/Dialog.js [hunk 6]
```diff
@@ -0,0 +1,16 @@
+import React from 'react';
+import classes from './Dialog.module.css';
+
+function Dialog(props) {
```
Comment (rainasong) [inline forgettable-frontend/src/components/Dialog/Dialog.js:4]: Can we please change 'Dialog' to 'Modal' as it will be consistent with the issue.

**Review by rainasong — CHANGES_REQUESTED** [review]
Hey Jiayou! Thanks for your hard work!
Please have a look at the comment I made above and if possible, make some changes to cover those clarified acceptance criteria. Sorry it wasn't very clear in the issue!
Thanks!

**Comment by rainasong** [comment]
Updated Modal component (name changed to CustomModal) to be able to accept children components. Styles are also able to be passed in from outside by specifying className variable.
By specifying 'hasCancel', 'hasConfirm' can add the default cancel or confirm button. If 'hasConfirm' is specified, 'onConfirm' function should be passed in as well.
Users are able to click anywhere outside the Modal to close it, or they can click on the Cancel button.

Currently this component can taking following props:
`CustomModal.propTypes = {
  open: PropTypes.bool.isRequired,
  hasCancel: PropTypes.bool,
  hasConfirm: PropTypes.bool,
  onClose: PropTypes.func.isRequired,
  onConfirm: PropTypes.func,
  className: PropTypes.string,
};`

Examples as below:
Screen Shot 2022-03-15 at 10 38 34 AM

**Review by rainasong — APPROVED** [review]

**Review by noahjade — APPROVED** [review]
Thanks for your work on this guys!!

**Review by hajineats — APPROVED** [review]

**Pull request merged**